grpc-ballerina/auth_listener_file_user_store_basic_auth_handler.bal

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/auth;
+
+# Represents file user store configurations for Basic Auth authentication.
+public type FileUserStoreConfig record {|
+    *auth:FileUserStoreConfig;
+|};
+
+# Defines the file store Basic Auth handler for listener authentication.
+public class ListenerFileUserStoreBasicAuthHandler {
+
+    auth:ListenerFileUserStoreBasicAuthProvider provider;
+
+    # Initializes the `http:ListenerFileUserStoreBasicAuthHandler` object.
+    #
+    # + config - The `http:FileUserStoreConfig` instance
+    public isolated function init(FileUserStoreConfig config = {}) {
+        self.provider = new (config);
+    }
+
+    # Authenticates with the relevant authentication requirements.
+    #
+    # + headers - The headers map `map<string|string[]>` as an input
+    # + return - The `auth:UserDetails` instance or else an `UnauthenticatedError` error
+    public isolated function authenticate(map<string|string[]> headers) returns auth:UserDetails|UnauthenticatedError {
+        string|Error credential = extractCredential(headers);
+        if (credential is Error) {
+            return error UnauthenticatedError(credential.message());
+        } else {
+            auth:UserDetails|auth:Error details = self.provider.authenticate(credential);
+            if (details is auth:Error) {
+                return error UnauthenticatedError(details.message());
+            }
+            return checkpanic details;
```

Comment:
Shall we remove the `checkpanic` usages? Maybe we can fix this from a separate PR, if it is there in all over the module. Remove the same usages in HTTP module with this PR https://github.com/ballerina-platform/module-ballerina-http/pull/282

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/268/commits/5496aaed70eb2790f7f2eff2a69a41fcab753f6d